### PR TITLE
Fixes #12. 

### LINKED ISSUE #12: Test violations codes and docs
We need to tests that all violations 

1. have unique codes
2. correctly sorted inside the code
3. have docs in the module docstrings

You can have a look at the example (this code can be even copy-pasted): https://github.com/wemake-services/wemake-python-styleguide/tree/master/tests/test_violations

### PR DESCRIPTION
**Pull request by kaelscion**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,9 +19,6 @@ We follow Semantic Versions.
 
 
 ## Version 0.1.2
-
```
Comment (sobolevn) [inline CHANGELOG.md]: Probably an accident 🙂 
There's no need to edit existing CHANGELOG entries, only add new ones

**Comment by sobolevn** [comment]
Please, fix the CI and let's merge it.
Ping me if you need my help with it.

**Comment by sobolevn** [comment]
@kaelscion friendly ping, how's going for you?

**Comment by kaelscion** [comment]
Thanks for checking in! I'm afraid I am not familiar enough with Travis CI to know what I am doing wrong. Would there be a way for you to take a look and help me understand what process is failing so that I can fix it?

**Comment by sobolevn** [comment]
@kaelscion sure! There several things to fix (they are not related to TravisCI, just ordinary test failures):

## Wrong fixture

```
 fixture 'all_module_violations' not found
>       available fixtures: all_violations, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, fixture_path, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
```

Is raised because `all_module_violations` is not found. I guess that you have meant `all_violations`.

## Failing test_checker_default_locations

This test fails: https://github.com/wemake-services/dotenv-linter/pull/88/files#diff-638e0e1892e2b7bf3dc040ab8e0867d7R17

Why? Because it does not do anything.

```
    def location(self) -> int:
        """Returns in-file location of a violation."""
>       raise NotImplementedError('Should be redefined in a subclass')
E       NotImplementedError: Should be redefined in a subclass
```

It should be something like `with pytest.raises(NotImplementedError):`

## Module has no lineno

```
________________________ test_visitor_returns_location _________________________
    def test_visitor_returns_location():
        """Ensures that `BaseNodeVisitor` return correct violation message."""
        visitor = BaseFSTViolation(node=ast.parse(''), text='violation')
        visitor.error_template = '{0}'
        visitor.code = 1
>       assert visitor.location() == (0, 0, 'Z001 violation')
tests/test_violations/test_implementation.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <dotenv_linter.violations.base.BaseFSTViolation object at 0x7ff43aadcb00>
    @final
    def location(self) -> int:
        """Returns in-file location of a violation."""
>       return self._node.lineno
E       AttributeError: 'Module' object has no attribute 'lineno'
```

This one is tricky. I have missed `lineno` on `Module` instance.

**Comment by kaelscion** [comment]
@sobolevn thank you so much for pointing that out! I will start addressing those issues soon as!

File: dotenv_linter/violations/base.py [hunk 2]
```diff
@@ -27,7 +28,9 @@ def as_line(self) -> str:
 
     def location(self) -> int:
         """Returns in-file location of a violation."""
-        raise NotImplementedError('Should be redefined in a subclass')
+        with pytest.raises(NotImplementedError):
```
Comment (sobolevn) [inline dotenv_linter/violations/base.py]: That should be placed inside a test, not in the implementation. See this example.
Test: https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_visitors/test_base.py#L14-L17
Source: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/base.py#L129

**Comment by kaelscion** [comment]
okay, so I've got to bite the bullet and ask for help. I cannot seem to make heads or tails of these issues and feel, at this point, that I'm only making the problems worse and more tests fail the more I pound away at it. Is there a time where we could do some pair programming or have a conversation about how to address these issues? I know I can't see the forest through the trees and need some help. Let me know and thank you so much!

**Comment by sobolevn** [comment]
@kaelscion sure, we can schedule a call this week. I am free since wednesday.

**Comment by kaelscion** [comment]
@sobolevn do you have any time available on Thursday or is Wednesday better? Also do you prefer Skye or Zoom?

**Comment by sobolevn** [comment]
Both are fine, but wed is slightly better. Let's use Skype, since Zoom is unsafe. See https://medium.com/bugbountywriteup/zoom-zero-day-4-million-webcams-maybe-an-rce-just-get-them-to-visit-your-website-ac75c83f4ef5

**Comment by kaelscion** [comment]
Great! I'm not sure of the time exchange but does early afternoon, between 12:30-1 EST work for you?

**Comment by sobolevn** [comment]
Снимок экрана 2019-07-17 в 9 15 27

Yeap, that works for me.

**Pull request closed without merging**